src/components/network-map-viewer/network/network-map.jsx

```diff
@@ -523,6 +528,118 @@ const NetworkMap = (props) => {
+        props.onFeaturesChanged(features);
+    }, [features]);
+
+    const onUpdate = useCallback((e) => {
+        setFeatures((currFeatures) => {
+            const newFeatures = { ...currFeatures };
+            for (const f of e.features) {
+                newFeatures[f.id] = f;
+            }
+            return newFeatures;
+        });
+    }, []);
+
+    const getPolygonFeatures = () => features;
+    const getSelectedSubstations = () => {
+        const substations = getSubstationsInPolygon(
+            features,
+            props.mapEquipments,
+            props.geoData
+        );
+        return (
+            substations.filter((substation) => {
+                return substation.voltageLevels.some((vl) =>
+                    props.filteredNominalVoltages.includes(vl.nominalV)
+                );
+            }) ?? []
+        );
+    };
+
+    const getSelectedVoltageLevel = () => {
+        const selectedVL = getVoltageLevelFromSubstation(
+            getSelectedSubstations()
+        );
+        return selectedVL.filter((vl) =>
+            props.filteredNominalVoltages.includes(vl.nominalV)
+        );
+    };
+
+    function getSelectedLinesInPolygon(
+        network,
+        lines,
+        geoData,
+        polygonCoordinates
+    ) {
+        return lines.filter((line) => {
+            try {
+                const linePos = geoData.getLinePositions(network, line);
+                if (!linePos) {
+                    return false;
+                }
+
+                return (
+                    booleanPointInPolygon(linePos[0], polygonCoordinates) &&
+                    booleanPointInPolygon(linePos[1], polygonCoordinates)
+                );
+            } catch (error) {
+                return false;
+            }
+        });
+    }
+    const getSelectedLines = () => {
+        //check if polygon is defined correctly
+        const firstPolygonFeatures = Object.values(features)[0];
+        const polygonCoordinates = firstPolygonFeatures?.geometry;
+        if (!polygonCoordinates || polygonCoordinates.coordinates < 3) {
+            return [];
+        }
+        //for each line, check if it is in the polygon
+        const selectedLines = getSelectedLinesInPolygon(
+            props.mapEquipments,
+            mapEquipmentsLines,
+            props.geoData,
+            polygonCoordinates
+        );
+        return filterLinesByNominalVoltages(selectedLines);
+    };
+
+    const filterLinesByNominalVoltages = (equipements) => {
+        return equipements.filter((equipement) => {
+            const isVL1 = props.filteredNominalVoltages.includes(
+                props.mapEquipments.getVoltageLevel(equipement.voltageLevelId1)
+                    .nominalV
+            );
+            const isVL2 = props.filteredNominalVoltages.includes(
+                props.mapEquipments.getVoltageLevel(equipement.voltageLevelId2)
+                    .nominalV
+            );
+            return isVL1 || isVL2;
```

Comment:
```suggestion
            return props.filteredNominalVoltages.includes(
                props.mapEquipments.getVoltageLevel(equipement.voltageLevelId1)
                    .nominalV
            ) || props.filteredNominalVoltages.includes(
                props.mapEquipments.getVoltageLevel(equipement.voltageLevelId2)
                    .nominalV
            );
```

Which is optimized because if VL1 is included then don't need to check for VL2